torch/testing/_internal/common_methods_invocations.py

```diff
@@ -2154,6 +2154,32 @@ def sample_inputs_hardswish(self, device, dtype, requires_grad):
                requires_grad=requires_grad, low=-5, high=5)) for _ in range(1, N)]
     return tensors
 
+def sample_inputs_linear(has_bias, self, device, dtype, requires_grad):
+    features_options = [[3, 4], [128, 128]]
+    batch_options = [
+        [], # no batch
+        [64],
+        [5, 7],
+    ]
+
+    sample_inputs = []
+    for (in_feat, out_feat), batch_shape in itertools.product(features_options, batch_options):
+        input_tensor = make_tensor(batch_shape + [in_feat], device=device,
```

Comment:
Reminder that as a style thing if this partials `make_tensor` it would be more succinct:

```
make_arg = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad, low=-2, high=2)

...

input_tensor = make_arg(batch_shape + [in_feat])
weight = make_arg([out_feat, in_feat])
...
```

Reply:
Thanks for the reminder! partial make it a lot more succinct